netaddr.go

```diff
@@ -1066,13 +1060,19 @@ func (p IPPrefix) Contains(ip IP) bool {
 		return false
 	}
 	if ip.Is4() {
-		m := mask4(p.Bits)
-		return uint32(ip.addr.lo)&m == uint32(p.IP.addr.lo)&m
+		// xor the IP addresses together; mismatched bits are now ones.
+		// Shift away the number of bits we don't care about.
+		// We know that p.Bits is in the range 0..32 because p is Valid;
+		// the compiler doesn't know that, so mask with 63 to help it.
+		// Now truncate to 32 bits, because this is IPv4.
+		// If all the bits we care about are equal, the result wil be zero.
+		return uint32((ip.addr.lo^p.IP.addr.lo)>>((32-p.Bits)&63)) == 0
```

Comment:
I do want to call out that this feels significantly less readable than the prior code, for a 6% gain. However, this is also my kind of IP address manipulation code, so I'm all for it :P